src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,65 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+
+	private String example; 
+    private String validationRegex;
+	
+	protected String value;
+    protected boolean isPrivate;
+	
+    public Contact() {}
+    
+    /**
+     * Validates given contact information.
+     *
+     * @throws IllegalValueException if given value string is invalid.
+     */
+    public Contact(String value, boolean isPrivate, 
+    		       String example, String messageAddressConstraints, 
+    		       String validationRegex) throws IllegalValueException{
+        if (!isValid(value)) {
+            throw new IllegalValueException(messageAddressConstraints);
+        }
+        this.value = value;
+        this.isPrivate = isPrivate;
+        this.example = example;
+        this.validationRegex = validationRegex;
+    }
+    
+    
+    /**
+     * Returns true if a given string is a valid person email.
+     */
+    public boolean isValid(String test){
+    	return test.matches(this.validationRegex);
+    }
+    
+    public String getValue() {
+    	return this.value;
+    }
+    
+    public boolean isPrivate() {
+        return isPrivate;
+    }
+    
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
```

Comment:
aah! can you see the problem here? you can now potentially match `Address` with `Phone`; why?